src/peripheral/dwt.rs

```diff
@@ -356,12 +373,27 @@ pub struct ComparatorAddressSettings {
     pub access_type: AccessType,
 }
 
+/// Settings for cycle count matching
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+pub struct CycleCountSettings {
+    /// The function selection used.
+    /// See Table C1-15 for DWT cycle count comparison functions.
+    pub emit: EmitOption,
+    /// The cycle count value to compare against.
+    pub compare: u32,
+}
+
 /// The available functions of a DWT comparator.
 #[derive(Debug, Eq, PartialEq, Copy, Clone)]
 #[non_exhaustive]
 pub enum ComparatorFunction {
     /// Compare accessed memory addresses.
     Address(ComparatorAddressSettings),
+    /// Compare cycle count & target value.
+    ///
+    /// **NOTE**: only supported by comparator 0 and if the HW supports the cycle counter.
+    /// Check `dwt.has_cycle_counter` for support. See C1.8.1 for more details.
```

Comment:
```suggestion
    /// Check [DWT::has_cycle_counter] for support. See C1.8.1 for more details.
```

Reply:
I had initially tried this but rust-analyzer didn't highlight it as valid, however checking `cargo doc` looks like it can. Fixed, but using "`"'s for fixed-width text formatting.

Reply:
It is a valid link then? Mustn't you use the below?

    /// [`DWT::has_cycle_counter`](DWT::has_cycle_counter)

Reply:
[`DWT::has_cycle_compare`] wasn't highlighting for some reason so I had dropped the square brackets. Seems like just a bug in the highlighter or rust-analyzer 🤷